cpp_utils/WiFi.cpp

```diff
@@ -42,13 +42,27 @@ static void setDNSServer(char *ip) {
 */
 
 
-WiFi::WiFi() {
-    ip      = "";
-    gw      = "";
-    netmask = "";
+}
+*/
+
+/**
+ * @brief Creates and uses a default event handler
+ */
+WiFi::WiFi()
+    : ip("")
+    , gw("")
+    , netmask("")
+    , wifiEventHandler(nullptr)
+{
     wifiEventHandler = new WiFiEventHandler();
 }
 
+/**
+ * @brief Deletes the event handler that was used by the class
+ */
+WiFi::~WiFi() {
+    delete wifiEventHandler;
```

Comment:
Do we need to check that wifiEventHandler isn't null before deleting?

Reply:
No, C++ does that automatically. If it's null delete just does nothing, so the if is irrelevant.